Docs/migration-guide-3.x-4.x.md

```diff
@@ -0,0 +1,50 @@
+## 3.x to 4.x Migration Guide
+
+4.0 SDK introduced a new call state `RECONNECTING`. You will need to update any logic you have implemented that relies on the call state. The simplest approach is to treat a `RECONNECTING` just like a `CONNECTED` and keep the current behavior.
+
+4.0 has a new state `RECONNECTING` in `Call.State` and two new callbacks `onReconnecting(...)`, `onReconnected(...)` in `Call.Listener()`. Any prior implementation of `Call.Listener()` will need to be updated with the new callbacks.
+
+```
+private Call.Listener callListener() {
+    return new Call.Listener() {
+    
+        @Override
+        public void onRinging(NonNull Call call) {
```

Comment:
s/NonNull/`@`NonNull

This applies to `onConnectFailure` and `onConnected` callbacks.